docs/page-objects/inputs.rst

```diff
@@ -0,0 +1,65 @@
+.. _inputs:
+
+======
+Inputs
+======
+
+:ref:`Page object classes <page-object-classes>`, in their ``__init__`` method,
+must define input parameters with type hints pointing to input classes.
+
+Those input classes may be:
+
+-   :ref:`Built-in web-poet input classes <built-in-inputs>`.
+
+-   :ref:`Custom input classes <custom-inputs>`.
+
+-   Other :ref:`page object classes <page-object-classes>`.
+
+-   :ref:`Item classes <items>`, when using a :ref:`framework <frameworks>`
+    that can provide item classes.
+
+Based on the target URL and parameter type hints, :ref:`frameworks
+<frameworks>` automatically build the required objects at run time, and pass
+them to the ``__init__`` method of the corresponding page object class.
+
+For example, if a page object class has an ``__init__`` parameter of type
+:class:`~web_poet.page_inputs.HttpResponse`, and the target URL is
```

Comment:
The link doesn't work for some reason.

Reply:
https://github.com/scrapinghub/web-poet/pull/152/commits/b958a1a25cc9d04a9d16de4b5f114fc74c44d480